community_server/src/Model/Table/TransactionsTable.php

```diff
@@ -122,4 +123,147 @@ public function buildRules(RulesChecker $rules)
 
         return $rules;
     }
+    
+    public function sortTransactions($a, $b)
+    {
+        if ($a['date'] == $b['date']) {
+            return 0;
+        }
+        return ($a['date'] > $b['date']) ? -1 : 1;
+    }
+
+    
+    public function listTransactionsHumanReadable($stateUserTransactions, array $user, $decay = true) 
+    {
+        
+        $stateUsersTable    = TableRegistry::getTableLocator()->get('StateUsers');
+        $stateBalancesTable = TableRegistry::getTableLocator()->get('StateBalances');
+        
+        $transaction_ids = [];
+        $involved_user_ids = [];
+        $stateUserTransactionsCount = 0;
+        foreach($stateUserTransactions as $su_transaction) {
```

Comment:
I do not understand why this is not calling the `listTransactions()` ?
Does that not exist? Can't we reuse some code here?

Reply:
This is called by the new listTransactions. 
I have rewritten listTransactions to finally able to use paging with using state_user_transactions.
Some Code I have moved in this function in Model/Table/TransactionsTable to able to call it from different Controllers. 
So I can also use this from old StateBalances/overview in old frontend but I haven't replaced it yet.

The new approach is much faster than the old one, I think it needed more than the half time less then bevore.

Reply:
Great! I was just wondering why there is a "...HumanReadable" function with alot of code. If it makes sense its totally fine with